src/cryptoadvance/specter/wallet.py

```diff
@@ -915,7 +919,14 @@ def check_utxo(self):
                                 utxo_item["txid"] == utxo_txid
                                 and utxo_item["vout"] == utxo_vout
                             ):
-                                tx_copy["amount"] = utxo_item["amount"]
+                                if utxo_item.get("amount"):
+                                    amount = utxo_item.get("amount")
+                                else:
+                                    # in the case of locked amounts, the stupid listlockunspent call does not contain reasonable utxo-data
+                                    amount = self.rpc.gettransaction(tx_copy["txid"])[
+                                        "details"
+                                    ][utxo_item["vout"]]["amount"]
```

Comment:
The vout can not be used as index as vout is a key in the elements of the details array. See:
https://developer.bitcoin.org/reference/rpc/gettransaction.html?highlight=gettransaction

Reply:
It most probably can be used as an index. This will hopefully soon be clarified here: https://bitcoin.stackexchange.com/questions/117597/order-of-the-details-array-in-the-output-of-the-gettransaction-wallet-api-call